### PR TITLE
Typescript

### PR DESCRIPTION
**Pull request by markoffk**

### CODE REVIEW

File: exmg-paper-token-input.ts [hunk 1]
```diff
@@ -0,0 +1,546 @@
...
+
+  @query('#inputWidthHelper')
+  private inputWidthHelperNode?: HTMLElement;
+
+  @query('#menu')
+  private menuElement?: PaperMenuButton;
+
+  constructor() {
+    super();
+    this.onIronInputKeyDown = this.onIronInputKeyDown.bind(this);
```
Comment (pgoc) [inline exmg-paper-token-input.ts:158]: You don't need to bind everything to this only such functions which you use as handlers fro manual added listeners

File: exmg-paper-token-input.ts [hunk 2]
```diff
@@ -0,0 +1,546 @@
...
+
+  private indexOfValue(value: SelectedValue): number {
+    return this.listBoxNode!.items!
+        .map((item: HTMLElement) => this.getPaperItemValue(item))
+        .indexOf(value);
+  }
+
+  /** EVENT HANDLERS */
+
+  private previousClickWasInside: boolean = false;
```
Comment (pgoc) [inline exmg-paper-token-input.ts]: isn't better to keep properties on top of class ?

File: exmg-paper-token-input.ts [hunk 3]
```diff
@@ -0,0 +1,546 @@
...
+      },
+      bubbles: true,
+      composed: true,
+    }));
+  }
+
+  private emitItemDeselectEvent(value: SelectedValue, item: HTMLElement): void {
+    this.dispatchEvent(new CustomEvent('exmg-token-input-deselect', {
+      detail: {
+        value,
```
Comment (pgoc) [inline exmg-paper-token-input.ts:279]: Seems that detail is not necessary or at least value item is undefined ?

File: package.json [hunk 4]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "@exmg/exmg-paper-token-input",
   "flat": true,
-  "version": "3.0.3",
+  "version": "4.0.0",
```
Comment (pgoc) [inline package.json]: we should increase version via command npm version after merging to master

File: tslint.json [hunk 5]
```diff
@@ -0,0 +1,8 @@
+{
+  "extends": [
+    "tslint-config-exmg"
+  ],
+  "rules": {
+    "ter-prefer-arrow-callback": [false]
```
Comment (pgoc) [inline tslint.json]: this rule is only for tests shouldn't be here

**Review by pgoc — APPROVED** [review]

File: exmg-paper-token-input.ts (hunk 1 above)
Comment (markoffk) [reply exmg-paper-token-input.ts:158]: done

File: exmg-paper-token-input.ts (hunk 2 above)
Comment (markoffk) [reply exmg-paper-token-input.ts]: done

File: package.json (hunk 4 above)
Comment (markoffk) [reply package.json]: done

File: tslint.json (hunk 5 above)
Comment (markoffk) [reply tslint.json]: done

File: exmg-paper-token-input.ts (hunk 3 above)
Comment (markoffk) [reply exmg-paper-token-input.ts:279]: no need

**Review by marksmits — APPROVED** [review]

**Pull request merged**